### PR TITLE
1493 remove event from activity log and add a from and to field for better translatability

### LINKED ISSUE #1493: Remove `event` from Activity Log and add a `from` and `to` field for better translatability
## Is your feature request related to a problem? Please describe 👀
The events for stock activity logs are currently hard coded in English in the backend.
image

We would also like to log sensor location changes from `location a` to `location b` and there will probably be more use cases for this in the future too, so nice to have it done now.

## Describe the solution you'd like 🎁
- [ ] Add a `from` and `to ` field to `activity_log`.
- [ ] Migrate current `events` column data to new `from` field. 
- [ ] Remove `events` column from `activity_log`
- [ ] Rename `activity_log_stock_entry` to be more generic and refactor.
- [ ] Refactor frontend to apply above changes

### Describe alternatives you've considered 💭
Using emojis 🍡

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #1493 

# 👩🏻‍💻 What does this PR do? 
Adds two new fields to the activity log for better translatability: `change_from` and `change_to`. Migrates the existing `event` column to `change_from` and remove it. Updated this change through backend and frontend.

# 🧪 How has/should this change been tested? 
- [ ] Go to stock line detail view
- [ ] Update any of the fields in stock line
- [ ] Go to log (currently no other translation for this other than English so....)

## 💌 Any notes for the reviewer?
Current mSupply PR for this change https://github.com/openmsupply/msupply/pull/13752

### CODE REVIEW

File: server/repository/src/migrations/v1_04_00/activity_log.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use crate::{migrations::sql, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    sql!(
+        connection,
+        r#"
+        ALTER TABLE activity_log ADD COLUMN change_to TEXT;
```
Comment (roxy-dao) [inline server/repository/src/migrations/v1_04_00/activity_log.rs:7]: `to` and `from` are reserved, so have used alternative

File: server/service/src/stock_line/update.rs [hunk 2]
```diff
@@ -279,21 +279,21 @@ fn log_stock_changes(
...
     }
     if existing.on_hold != new.on_hold && new.on_hold {
-        activity_log_stock_entry(
+        activity_log_entry(
             &ctx,
             ActivityLogType::StockOnHold,
             Some(new.id.to_owned()),
-            Some("off hold".to_string()),
-            Some("on hold".to_string()),
+            Some("\u{2705}".to_string()), // tick unicode
```
Comment (roxy-dao) [inline server/service/src/stock_line/update.rs:286]: Wasn't too sure what kind of unicode to use or which way around they should be... even emojis can be country-specific! 😭

File: server/repository/src/migrations/v1_04_00/activity_log.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        ALTER TABLE activity_log ADD COLUMN change_from TEXT;
+        UPDATE activity_log SET change_from = event;
+        ALTER TABLE activity_log DROP COLUMN event;
```
Comment (roxy-dao) [inline server/repository/src/migrations/v1_04_00/activity_log.rs:10]: Actually... could have probably just renamed this.....

File: server/service/src/stock_line/update.rs [hunk 4]
```diff
@@ -279,21 +279,21 @@ fn log_stock_changes(
...
     }
     if existing.on_hold != new.on_hold && new.on_hold {
-        activity_log_stock_entry(
+        activity_log_entry(
             &ctx,
             ActivityLogType::StockOnHold,
             Some(new.id.to_owned()),
-            Some("off hold".to_string()),
-            Some("on hold".to_string()),
+            None,
```
Comment (roxy-dao) [inline server/service/src/stock_line/update.rs:286]: The log type already specifies what is happening to the stock line

File: server/repository/src/migrations/v1_04_00/activity_log.rs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+use crate::{migrations::sql, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    sql!(
+        connection,
+        r#"
+        ALTER TABLE activity_log ADD COLUMN change_to TEXT;
+        ALTER TABLE activity_log RENAME COLUMN event TO change_from;
```
Comment (roxy-dao) [inline server/repository/src/migrations/v1_04_00/activity_log.rs:8]: Just renaming so all the information in previous event column will still exist

**Review by mark-prins — CHANGES_REQUESTED** [review]
The translation / sync needs a column rename.. otherwise, it's all good! works nicely thanks 😄 

I've put the rename into a [suggestion branch](https://github.com/openmsupply/open-msupply/compare/1493-Remove-event-from-Activity-Log-and-add-a-from-and-to-field-for-better-translatability...1493-suggestion) in case that's helpful

File: server/service/src/sync/translations/activity_log.rs [hunk 6]
```diff
@@ -65,7 +67,8 @@ impl SyncTranslation for ActivityLogTranslation {
             store_id: Some(data.store_id),
             record_id: Some(data.record_id),
             datetime: data.datetime,
-            event: data.event,
+            change_to: data.change_to,
```
[the remark this answers is not in the document]
Comment (roxy-dao) [reply server/service/src/sync/translations/activity_log.rs:70]: Ah right! I'm so inconsistent in writing issues 🤣

**Review by mark-prins — APPROVED** [review]
👍  thanks for the changes LGTM!

**Pull request merged**